spec/where_spec.cr

```diff
@@ -0,0 +1,23 @@
+require "./spec_helper"
+
+macro should_negate(original_where, expected_negated_where)
+  clause = {{original_where}}.new("column", "value").negated
+  clause.column.should eq "column"
+  clause.value.should eq "value"
+  clause.should be_a({{expected_negated_where}})
+end
+
+describe LuckyRecord::Where do
+  it "can be negated" do
+    should_negate(LuckyRecord::Where::Equal, LuckyRecord::Where::NotEqual)
+    should_negate(LuckyRecord::Where::NotEqual, LuckyRecord::Where::Equal)
+    should_negate(LuckyRecord::Where::GreaterThan, LuckyRecord::Where::LessThanOrEqualTo)
+    should_negate(LuckyRecord::Where::GreaterThanOrEqualTo, LuckyRecord::Where::LessThan)
+    should_negate(LuckyRecord::Where::LessThan, LuckyRecord::Where::GreaterThanOrEqualTo)
+    should_negate(LuckyRecord::Where::LessThanOrEqualTo, LuckyRecord::Where::GreaterThan)
+    should_negate(LuckyRecord::Where::Like, LuckyRecord::Where::NotLike)
+    should_negate(LuckyRecord::Where::NotLike, LuckyRecord::Where::Like)
+    should_negate(LuckyRecord::Where::Ilike, LuckyRecord::Where::NotIlike)
+    should_negate(LuckyRecord::Where::NotIlike, LuckyRecord::Where::Ilike)
```

Comment:
Woah, I could not make your suggestion work for the life of me. As soon as I put the `should be_a()` into a `private def` it fails to compile with 

```
macro didn't expand to a valid program, it expanded to:

================================================================================
--------------------------------------------------------------------------------
   1.       Spec::BeAExpectation(expected_negated_where).new
   2.
--------------------------------------------------------------------------------
Syntax error in expanded macro: be_a:1: expecting token 'CONST', not 'expected_negated_where'

      Spec::BeAExpectation(expected_negated_where).new
```
I sure might be doing something wrong, or am I stumbling on a known issue (https://github.com/crystal-lang/crystal/issues/236)? Solved it with a macro as well, not really sure on the pitfalls/benefits at this point. At least the specs are less boilerplate haha.

Reply:
Ah this makes sense. I didn't realize `be_a` is a macro so this works 👍